search/django/weedid/utils.py

```diff
@@ -0,0 +1,55 @@
+import os
+import re
+import json
+from django.core.files.storage import FileSystemStorage
+from weedid.models import Dataset, WeedidUser
+
+
+def store_tmp_image(image, image_dir):
+    fs = FileSystemStorage()
+    fs.save(os.path.join(image_dir, image.name), image)
+
+
+def store_tmp_weedcoco(weedcoco, upload_dir):
+    fs = FileSystemStorage()
+    weedcoco_path = os.path.join(upload_dir, weedcoco.name)
+    fs.save(weedcoco_path, weedcoco)
+    return weedcoco_path
+
+
+def setup_upload_dir(upload_userid_dir):
+    if not os.path.isdir(upload_userid_dir):
+        os.mkdir(upload_userid_dir)
+        latest_upload_dir_index = 0
+    else:
+        latest_upload_dir_index = max(
+            [
+                int(dir.split("_")[-1])
+                for dir in os.listdir(upload_userid_dir)
+                if re.fullmatch(r"^upload_\d+$", dir)
```

Comment:
We're likely to occasionally clear out the upload dir... Does this ID need to be unique in the DB as well?
Do we use the same ID for the repository directory? Does it need to be unique there too?? How do we ensure all criteria are met (and ideally handle race conditions?) Using a UUID is an easy solution, albeit perhaps overkill.

Reply:
Yes, let's use UUID.

Reply:
Is there a proper way to test with random UUID?